src/main/java/uk/gov/pay/products/resources/ProductResource.java

```diff
@@ -64,21 +60,58 @@ public Response findProduct(@PathParam("productExternalId") String productExtern
                         Response.status(NOT_FOUND).build());
     }
 
+    @GET
+    @Path("/gateway-account/{gatewayAccountId}/products/{productExternalId}")
+    @Produces(APPLICATION_JSON)
+    @Consumes(APPLICATION_JSON)
+    public Response findProduct(@PathParam("gatewayAccountId") Integer gatewayAccountId, @PathParam("productExternalId") String productExternalId) {
+        logger.info("Find a product with externalId - [ {} ]", productExternalId);
+        return productFactory.productFinder().findByGatewayAccountIdAndExternalId(gatewayAccountId, productExternalId)
+                .map(product ->
+                        Response.status(OK).entity(product).build())
+                .orElseGet(() ->
+                        Response.status(NOT_FOUND).build());
+    }
+
+    // TODO: TO BE REMOVED!
+    // Leaving this provisionally for backward compatibility
     @PATCH
-    @Path(DISABLE_PRODUCT_RESOURCE_PATH)
+    @Path("/products/{productExternalId}/disable")
     @Produces(APPLICATION_JSON)
     @Consumes(APPLICATION_JSON)
-    public Response disableProduct(@PathParam("productExternalId") String productExternalId) {
+    public Response deprecatedDisableProduct(@PathParam("productExternalId") String productExternalId) {
```

Comment:
Shouldn't we add @Deprecated annotation as well?

Reply:
Addressed